srtcore/channel.h

```diff
@@ -236,14 +227,16 @@ class CChannel
             // IPv4 headers or IPv6 headers.
             if (cmsg->cmsg_level == IPPROTO_IP && cmsg->cmsg_type == IP_PKTINFO)
             {
-                in_pktinfo *dest_ip_ptr = (in_pktinfo*)CMSG_DATA(cmsg);
-                return sockaddr_any(dest_ip_ptr->ipi_addr, 0);
+                in_pktinfo dest_ip_ptr;
+                memcpy(&dest_ip_ptr, CMSG_DATA(cmsg), sizeof(struct in_pktinfo));
```

Comment:
```suggestion
                in_pktinfo dest_ip = *(in_pktinfo*)CMSG_DATA(cmsg);
```
And BTW, what is this change for? Is that because CMSG_DATA doesn't guarantee to return an even pointer?